File: classA/4/homework2/task2.py
import sys

from matrix import extract_matrix_from_csv, Matrix


def main():
    has_provided_filenames = len(sys.argv) > 2

    first_matrix_filename = sys.argv[1] if has_provided_filenames else './matrix1.csv'
    first_matrix_values = extract_matrix_from_csv(first_matrix_filename)
    first_matrix = Matrix(first_matrix_values)

    second_matrix_filename = sys.argv[2] if has_provided_filenames else './matrix2.csv'
    second_matrix_values = extract_matrix_from_csv(second_matrix_filename)
    second_matrix = Matrix(second_matrix_values)
    
    result = first_matrix * second_matrix

    if result:
        print('true' if result.is_asymmetric else 'false')
    else:
        print('undefined')

if __name__ == '__main__':
    main()

File: classB/02/homework/task2.rb
require "csv"

matrix1 = CSV.read("./matrix1.csv");
matrix2 = CSV.read("./matrix2.csv");

if matrix1.length != matrix2.length || matrix1[0].length != matrix2[0].length
	p 'undefined'
	return
end

rows = matrix1.length
cols = matrix1[0].length
if cols != rows
	p false
	return
end
result = Array.new(rows, 0){Array.new(cols, 0)}

for i  in 0..rows - 1
	for j in 0..cols - 1
		for k in 0..rows - 1
			#print "matrix1: ",matrix1[i][k],"  matrix2:",matrix2[k][j],"\n"
			result[i][j] += matrix1[i][k].to_i * matrix2[k][j].to_i
		end
	end
end

for i  in 0..rows - 1
	for j in 0..cols - 1
		if result[i][j] != (-1 * result[j][i])
			p false
			return
		end
	end
end

p true
